File: frontend/src/pages/dashboard/ecommerce/components/equations.ts
/**
 * Equations Configuration
 * ------------------------
 * Author: Saif Eldin Ayman
 * Version: 2.0.1
 * Last Updated: 2025-09-18
 * Description: This file contains the structured configuration
 * for nutrition- and health-related equations used in the calculators.
 * Each equation includes its id, code, name, input definitions,
 * and optional system variations.
 */

export const equationsConfig = [
  {
    "id": 1,
    "code": "Body Mass Index",
    "name": "Body Mass Index (BMI)",
    "system": { "name": "inputs_system", "label": "Inputs System", "type": "select", "options": ["Metric (kg, m)", "Imperial (lb, in)"] },
    "inputs": [
      [
        { "name": "weight_kg", "label": "Weight (kg)", "type": "number" },
        { "name": "height_m", "label": "Height (m)", "type": "number" }
      ],
      [
        { "name": "weight_lb", "label": "Weight (lb)", "type": "number" },
        { "name": "height_in", "label": "Height (in)", "type": "number" }
      ]
    ]
  },
  {
    "id": 2,
    "code": "Ideal Body Weight by Hamwi",
    "name": "Ideal Body Weight (Hamwi)",
    "inputs": [
      { "name": "gender", "label": "Gender", "type": "select", "options": ["male", "female"] },
      { "name": "height_cm", "label": "Height (cm)", "type": "number" }
    ]
  },
  {
    "id": 3,
    "code": "Ideal Body Weight by Lemmens",
    "name": "Ideal Body Weight (Lemmens)",
    "inputs": [
      { "name": "height_m", "label": "Height (m)", "type": "number" },
      { "name": "factor", "label": "Scaling Factor", "type": "number" }
    ]
  },
  {
    "id": 4,
    "code": "percent ideal body weight",
    "name": "Percent of Ideal Body Weight",
    "inputs": [
      { "name": "weight_kg", "label": "Weight (kg)", "type": "number" },
      { "name": "gender", "label": "Gender", "type": "select", "options": ["male", "female"] },
      { "name": "height_cm", "label": "Height (cm)", "type": "number" }
    ]
  },
  {
    "id": 5,
    "code": "adjusted body weight",
    "name": "Adjusted Body Weight",
    "inputs": [
      { "name": "weight_kg", "label": "Weight (kg)", "type": "number" },
      { "name": "gender", "label": "Gender", "type": "select", "options": ["male", "female"] },
      { "name": "height_cm", "label": "Height (cm)", "type": "number" }
    ]
  },
  {
    "id": 6,
    "code": "dry weight",
    "name": "Dry Weight",
    "inputs": [
      { "name": "weight_kg", "label": "Weight (kg)", "type": "number" },
      {
        "name": "edema_or_ascites",
        "label": "Condition",
        "type": "select",
        "options": [
          "ascites_minimal",
          "ascites_moderate",
          "ascites_severe",
          "edema_minimal",
          "edema_moderate",
          "edema_severe"
        ]
      }
    ]
  },
  {
    "id": 7,
    "code": "Ideal Body Weight amputation",
    "name": "Ideal Body Weight (Amputation)",
    "inputs": [
      { "name": "gender", "label": "Gender", "type": "select", "options": ["male", "female"] },
      { "name": "height_cm", "label": "Height (cm)", "type": "number" },
      { "name": "percent_amputation", "label": "Percent Amputation (%)", "type": "number" }
    ]
  },
  {
    "id": 8,
    "code": "waist height ratio",
    "name": "Waist-to-Height Ratio",
    "inputs": [
      { "name": "waist_cm", "label": "Waist (cm)", "type": "number" },
      { "name": "height_cm", "label": "Height (cm)", "type": "number" },
      { "name": "gender", "label": "Gender", "type": "select", "options": ["male", "female"] }
    ]
  },
  {
    "id": 9,
    "code": "percent usual body weight",
    "name": "Percent of Usual Body Weight",
    "inputs": [
      { "name": "weight_kg", "label": "Weight (kg)", "type": "number" },
      { "name": "usual_weight_kg", "label": "Usual Weight (kg)", "type": "number" }
    ]
  },
  {
    "id": 10,
    "code": "percent weight change",
    "name": "Percent Weight Change",
    "inputs": [
      { "name": "current_weight_kg", "label": "Current Weight (kg)", "type": "number" },
      { "name": "usual_weight_kg", "label": "Usual Weight (kg)", "type": "number" }
    ]
  },
  {
    "id": 11,
    "code": "Ideal Body Weight for Spinal Cord Injury",
    "name": "Ideal Body Weight (Spinal Cord Injury)",
    "inputs": [
      { "name": "gender", "label": "Gender", "type": "select", "options": ["male", "female"] },
      { "name": "height_in_inches", "label": "Height (inches)", "type": "number" },
      { "name": "classification", "label": "Classification", "type": "select", "options": ["none", "paraplegia", "tetraplegia", "quadriplegia"] }
    ]
  },
  {
    "id": 12,
    "code": "demi span height",
    "name": "Demi-Span Height Estimate",
    "inputs": [
      { "name": "gender", "label": "Gender", "type": "select", "options": ["male", "female"] },
      { "name": "demi_span_cm", "label": "Demi-Span (cm)", "type": "number" }
    ]
  },
  {
    "id": 13,
    "code": "knee height estimate",
    "name": "Knee Height Estimate",
    "inputs": [
      { "name": "gender", "label": "Gender", "type": "select", "options": ["male", "female"] },
      { "name": "knee_height_cm", "label": "Knee Height (cm)", "type": "number" },
      { "name": "age_years", "label": "Age (years)", "type": "number" }
    ]
  },
  {
    "id": 14,
    "code": "harris benedict",
    "name": "Harris-Benedict Equation",
    "inputs": [
      { "name": "gender", "label": "Gender", "type": "select", "options": ["male", "female"] },
      { "name": "weight_kg", "label": "Weight (kg)", "type": "number" },
      { "name": "height_cm", "label": "Height (cm)", "type": "number" },
      { "name": "age", "label": "Age (years)", "type": "number" }
    ]
  },
  {
    "id": 15,
    "code": "mifflin st jeor",
    "name": "Mifflin-St Jeor Equation",
    "inputs": [
      { "name": "gender", "label": "Gender", "type": "select", "options": ["male", "female"] },
      { "name": "weight_kg", "label": "Weight (kg)", "type": "number" },
      { "name": "height_cm", "label": "Height (cm)", "type": "number" },
      { "name": "age", "label": "Age (years)", "type": "number" }
    ]
  },
  {
    "id": 16,
    "code": "total energy expenditure",
    "name": "Total Energy Expenditure",
    "inputs": [
      { "name": "hbe_kcal", "label": "HBE kcal/day", "type": "number" },
      { "name": "physical_factor", "label": "Physical Factor", "type": "number" },
      { "name": "stress_factor", "label": "Stress Factor", "type": "number" }
    ]
  },
  {
    "id": 17,
    "code": "Penn state",
    "name": "Penn State Equation",
    "inputs": [
      { "name": "mifflin_kcal", "label": "Mifflin-St Jeor kcal/day", "type": "number" },
      { "name": "tmax_c", "label": "Max Temp (°C)", "type": "number" },
      { "name": "ve_l_min", "label": "Minute Ventilation (L/min)", "type": "number" },
      {
        "name": "older_and_obese",
        "label": "Older & Obese",
        "type": "select",
        "options": ["true", "false"]
      }
    ]
  },
  {
    "id": 18,
    "code": "cunningham",
    "name": "Cunningham Equation",
    "inputs": [
      { "name": "weight_kg", "label": "Weight (kg)", "type": "number" },
      { "name": "percent_body_fat", "label": "Body Fat (%)", "type": "number" }
    ]
  },
  {
    "id": 19,
    "code": "baseline fluid bsa",
    "name": "Baseline Fluid (BSA Method)",
    "inputs": [
      { "name": "weight_kg", "label": "Weight (kg)", "type": "number" }
    ]
  },
  {
    "id": 20,
    "code": "baseline fluid standard",
    "name": "Baseline Fluid (Standard)",
    "inputs": [
      { "name": "weight_kg", "label": "Weight (kg)", "type": "number" }
    ]
  },
  {
    "id": 21,
    "code": "insulin to carb ratio",
    "name": "Insulin-to-Carbohydrate Ratio",
    "inputs": [
      { "name": "total_daily_dose_iu", "label": "Total Daily Dose (IU)", "type": "number" },
      { "name": "insulin_type", "label": "Insulin Type", "type": "select", "options": ["rapid", "short"] }
    ]
  },
  {
    "id": 22,
    "code": "insulin sensitivity",
    "name": "Insulin Sensitivity Factor",
    "inputs": [
      { "name": "total_daily_dose_iu", "label": "Total Daily Dose (IU)", "type": "number" },
      { "name": "insulin_type", "label": "Insulin Type", "type": "select", "options": ["rapid", "short"] }
    ]
  },
  {
    "id": 23,
    "code": "insulin initial dose",
    "name": "Initial Insulin Dose",
    "inputs": [
      { "name": "weight_kg", "label": "Weight (kg)", "type": "number" },
      { "name": "by_kg", "label": "By kg?", "type": "select", "options": ["true", "false"] }
    ]
  },
  {
    "id": 24,
    "code": "free water deficit",
    "name": "Free Water Deficit",
    "inputs": [
      { "name": "weight_kg", "label": "Weight (kg)", "type": "number" },
      { "name": "sodium_mmol_l", "label": "Sodium (mmol/L)", "type": "number" },
      { "name": "gender", "label": "Gender", "type": "select", "options": ["male", "female"] },
      { "name": "age_years", "label": "Age (years)", "type": "number" }
    ]
  },
  {
    "id": 25,
    "code": "nitrogen balance",
    "name": "Nitrogen Balance",
    "inputs": [
      { "name": "protein_intake_g", "label": "Protein Intake (g)", "type": "number" },
      { "name": "urine_urea_n_g", "label": "Urine Urea N (g)", "type": "number" }
    ]
  },
  {
    "id": 26,
    "code": "nutrition risk index",
    "name": "Nutrition Risk Index",
    "inputs": [
      { "name": "albumin_g_dl", "label": "Albumin (g/dL)", "type": "number" },
      { "name": "weight_kg", "label": "Weight (kg)", "type": "number" },
      { "name": "gender", "label": "Gender", "type": "select", "options": ["male", "female"] },
      { "name": "height_cm", "label": "Height (cm)", "type": "number" }
    ]
  },
  {
    "id": 27,
    "code": "total lymphocyte count",
    "name": "Total Lymphocyte Count",
    "inputs": [
      { "name": "percent_lymphocytes", "label": "Lymphocytes (%)", "type": "number" },
      { "name": "wbc_10e3", "label": "WBC (×10³/μL)", "type": "number" }
    ]
  },
  {
    "id": 28,
    "code": "growth velocity",
    "name": "Growth Velocity",
    "inputs": [
      { "name": "weight_t1", "label": "Weight at T1 (kg)", "type": "number" },
      { "name": "weight_t2", "label": "Weight at T2 (kg)", "type": "number" },
      { "name": "height_t1", "label": "Height at T1 (cm)", "type": "number" },
      { "name": "height_t2", "label": "Height at T2 (cm)", "type": "number" },
      { "name": "time1_day", "label": "Time 1 (days)", "type": "number" },
      { "name": "time2_day", "label": "Time 2 (days)", "type": "number" },
      { "name": "time1_week", "label": "Time 1 (weeks)", "type": "number" },
      { "name": "time2_week", "label": "Time 2 (weeks)", "type": "number" },
      { "name": "preterm", "label": "Preterm", "type": "boolean" },
      { "name": "weight_current", "label": "Current Weight (kg)", "type": "number" },
      { "name": "age_months", "label": "Age (months)", "type": "number" }
    ]
  },
  {
    "id": 29,
    "code": "HBE",
    "name": "Harris–Benedict (HBE)",
    "inputs": [
      { "name": "gender", "label": "Gender", "type": "select", "options": ["male", "female"] },
      { "name": "weight_kg", "label": "Weight (kg)", "type": "number" },
      { "name": "height_cm", "label": "Height (cm)", "type": "number" },
      { "name": "age", "label": "Age (years)", "type": "number" }
    ]
  },
  {
    "id": 30,
    "code": "BMR",
    "name": "Mifflin-St Jeor",
    "inputs": [
      { "name": "gender", "label": "Gender", "type": "select", "options": ["male", "female"] },
      { "name": "weight_kg", "label": "Weight (kg)", "type": "number" },
      { "name": "height_cm", "label": "Height (cm)", "type": "number" },
      { "name": "age", "label": "Age (years)", "type": "number" }
    ]
  },
  {
    "id": 31,
    "code": "Quick Method for Macronutrients & Fluid",
    "name": "Quick Method for Macronutrients & Fluid",
    "inputs": [
      { "name": "weight_kg", "label": "Weight (kg)", "type": "number" },
      { "name": "min_factor", "label": "Min Factor", "type": "number" },
      { "name": "max_factor", "label": "Max Factor", "type": "number" }
    ]
  },
  {
    "id": 32,
    "code": "Macronutrients Distribution",
    "name": "Macronutrient Distribution",
    "inputs": [
      { "name": "calories", "label": "Calories (kcal)", "type": "number" },
      { "name": "carb_pct", "label": "Carbohydrates (%)", "type": "number" },
      { "name": "protein_pct", "label": "Protein (%)", "type": "number" },
      { "name": "fat_pct", "label": "Fat (%)", "type": "number" }
    ]
  },
  {
    "id": 33,
    "code": "RDA Calorie needs",
    "name": "Recommended Dietary Allowance for Calories",
    "inputs": [
      { "name": "age", "label": "Age (years)", "type": "number" },
      { "name": "weight", "label": "Weight (kg)", "type": "number" },
      { "name": "gender", "label": "Gender", "type": "select", "options": ["male", "female"] }
    ]
  },
  {
    "id": 34,
    "code": "RDA Protein needs",
    "name": "Recommended Dietary Allowance for Protein",
    "inputs": [
      { "name": "age", "label": "Age (years)", "type": "number" },
      { "name": "weight", "label": "Weight (kg)", "type": "number" },
      { "name": "gender", "label": "Gender", "type": "select", "options": ["male", "female"] }
    ]
  },
  {
    "id": 35,
    "code": "schofield bmr",
    "name": "Schofield BMR",
    "inputs": [
      { "name": "age", "label": "Age (years)", "type": "number" },
      { "name": "gender", "label": "Gender", "type": "select", "options": ["male", "female"] },
      { "name": "weight_kg", "label": "Weight (kg)", "type": "number" },
      { "name": "height_cm", "label": "Height (cm)", "type": "number" },
      { "name": "physical_activity", "label": "Physical Activity", "type": "select", "options": ["sedentary", "lightly_active", "moderately_active", "very_active", "extra_active"] },
      { "name": "stress_factor", "label": "Stress Factor", "type": "select", "options": ["minor_surgery", "major_surgery", "skeletal_trauma", "blunt_trauma", "closed_head_injury", "mild_infection", "moderate_infection", "severe_infection", "starvation", "burns_lt_20", "burns_20_40", "burns_gt_40"] }
    ]
  },
  {
    "id": 36,
    "code": "catchup growth",
    "name": "Catch-up Growth",
    "inputs": [
      { "name": "age", "label": "Age (years)", "type": "number" },
      { "name": "gender", "label": "Gender", "type": "select", "options": ["male", "female"] },
      { "name": "weight_kg", "label": "Weight (kg)", "type": "number" },
      { "name": "height_cm", "label": "Height (cm)", "type": "number" }
    ]
  },
  {
    "id": 37,
    "code": "gestation adjusted age",
    "name": "Gestation-Adjusted Age (Correction Age)",
    "inputs": [
      { "name": "gestational_age_weeks", "label": "Gestational Age (weeks)", "type": "number" },
      { "name": "chronological_age_weeks", "label": "Chronological Age (weeks)", "type": "number" }
    ]
  },
  {
    "id": 38,
    "code": "preterm estimated requirement",
    "name": "Preterm Estimated Requirement",
    "inputs": [
      { "name": "weight_kg", "label": "Weight (kg)", "type": "number" }
    ]
  },
  {
    "id": 39,
    "code": "ireton jones ventilator",
    "name": "Ireton-Jones (Ventilator-Dependent)",
    "inputs": [
      { "name": "age", "label": "Age (years)", "type": "number" },
      { "name": "gender", "label": "Gender", "type": "select", "options": ["male", "female"] },
      { "name": "weight_kg", "label": "Weight (kg)", "type": "number" },
      { "name": "trauma", "label": "Trauma", "type": "select", "options": ["present", "absent"] },
      { "name": "burn", "label": "Burn", "type": "select", "options": ["present", "absent"] }
    ]
  },
  {
    "id": 40,
    "code": "Ireton jones Spontaneous",
    "name": "Ireton-Jones (Spontaneous)",
    "inputs": [
      { "name": "age", "label": "Age (years)", "type": "number" },
      { "name": "weight_kg", "label": "Weight (kg)", "type": "number" },
      { "name": "height_cm", "label": "Height (cm)", "type": "number" }
    ]
  },
  {
    "id": 41,
    "code": "Curreri burn",
    "name": "Curreri Burn",
    "inputs": [
      { "name": "age", "label": "Age (years)", "type": "number" },
      { "name": "gender", "label": "Gender", "type": "select", "options": ["male", "female"] },
      { "name": "weight_kg", "label": "Weight (kg)", "type": "number" },
      { "name": "tbsa_percentage", "label": "TBSA Burn (%)", "type": "number" }
    ]
  },
  {
    "id": 42,
    "code": "pregnancy energy needs",
    "name": "Energy Needs in Pregnancy by Age & BMI group",
    "inputs": [
      { "name": "age", "label": "Age (years)", "type": "number" },
      { "name": "prepregnancy_weight_kg", "label": "Pre-pregnancy Weight (kg)", "type": "number" },
      { "name": "height_cm", "label": "Height (cm)", "type": "number" },
      { "name": "trimester", "label": "Trimester", "type": "select", "options": ["first", "second", "third"] },
      { "name": "physical_activity", "label": "Physical Activity", "type": "select", "options": ["inactive", "low_active", "active", "very_active"] },
      { "name": "gestation_week", "label": "Gestation Week", "type": "number" }
    ]
  },
  {
    "id": 43,
    "code": "pregnancy simple addition",
    "name": "Energy Needs in Pregnancy",
    "inputs": [
      { "name": "calorie_requirement", "label": "Base Calorie Requirement", "type": "number" },
      { "name": "trimester", "label": "Trimester", "type": "select", "options": ["first", "second", "third"] }
    ]
  },
  {
    "id": 44,
    "code": "lactation energy needs",
    "name": "Energy Needs in Lactation",
    "inputs": [
      { "name": "calorie_requirement", "label": "Base Calorie Requirement (kcal/day)", "type": "number" },
      { "name": "lactation_period", "label": "Lactation Period", "type": "select", "options": ["first_6_months", "second_6_months"] }
    ]
  },
  {
    "id": 45,
    "code": "down syndrome calorie",
    "name": "Down Syndrome Calorie Requirement (Ages 5–12)",
    "inputs": [
      { "name": "age", "label": "Age (years)", "type": "number" },
      { "name": "gender", "label": "Gender", "type": "select", "options": ["male", "female"] },
      { "name": "height_cm", "label": "Height (cm)", "type": "number" }
    ]
  },
  {
    "id": 46,
    "code": "cerebral palsy calorie",
    "name": "Cerebral Palsy Calorie Requirement (Ages 5–12)",
    "inputs": [
      { "name": "age", "label": "Age (years)", "type": "number" },
      { "name": "height_cm", "label": "Height (cm)", "type": "number" },
      { "name": "physical_activity", "label": "Physical Activity", "type": "select", "options": ["mild_to_moderate", "severe_physical_restriction", "severe_restricted_activity", "athetoid_cerebral_palsy"] }
    ]
  },
  {
    "id": 47,
    "code": "Prader Willi calorie",
    "name": "Prader-willi Syndrome Calorie Requirement (Ages 5–12)",
    "inputs": [
      { "name": "height_cm", "label": "Height (cm)", "type": "number" },
      { "name": "goal", "label": "Goal", "type": "select", "options": ["maintain", "weight_loss"] }
    ]
  },
  {
    "id": 48,
    "code": "prognostic nutrition index",
    "name": "Prognostic Nutrition Index (PNI)",
    "inputs": [
      { "name": "albumin", "label": "Albumin (g/dL)", "type": "number" },
      { "name": "triceps_skin_fold", "label": "Triceps Skin Fold (mm)", "type": "number" },
      { "name": "transferrin", "label": "Transferrin (mg/dL)", "type": "number" },
      { "name": "delayed_skin_hypersensitivity", "label": "Delayed Skin Hypersensitivity (score)", "type": "number" }
    ]
  },
  {
    "id": 49,
    "code": "prognostic inflammatory nutrition index",
    "name": "Prognostic Inflammatory and Nutrition Index",
    "inputs": [
      { "name": "c_reactive_protein", "label": "C-Reactive Protein (mg/L)", "type": "number" },
      { "name": "alpha_1_acid_glycoprotein", "label": "Alpha-1 Acid Glycoprotein (g/L)", "type": "number" },
      { "name": "prealbumin", "label": "Prealbumin (g/L)", "type": "number" },
      { "name": "albumin", "label": "Albumin (g/L)", "type": "number" }
    ]
  }
  
];

File: backend/nutrition/management/commands/equations.py
from django.core.management.base import BaseCommand
from nutrition.models import Equation

EQUATIONS = [
    {
        "name": "Body Mass Index",
        "code": "bmi",
        "function_path": "nutrition.utils.bmi",
        "description": "BMI: weight/height² (metric or imperial), with risk categories."
    },
    {
        "name": "Hamwi IBW",
        "code": "ibw_hamwi",
        "function_path": "nutrition.utils.ibw_hamwi",
        "description": "Ideal Body Weight using Hamwi formula, based on gender and height."
    },
    {
        "name": "Lemmens IBW",
        "code": "ibw_lemmens",
        "function_path": "nutrition.utils.ibw_lemmens",
        "description": "Ideal Body Weight using Lemmens equation: factor × height²."
    },
    {
        "name": "Percent IBW",
        "code": "percent_ibw",
        "function_path": "nutrition.utils.percent_ibw",
        "description": "Current weight as % of IBW with nutritional interpretation."
    },
    {
        "name": "Adjusted Body Weight",
        "code": "adjusted_body_weight",
        "function_path": "nutrition.utils.adjusted_body_weight",
        "description": "Adjusted body weight for obese patients: IBW + 0.4 × (ABW-IBW)."
    },
    {
        "name": "Dry Weight",
        "code": "dry_weight",
        "function_path": "nutrition.utils.dry_weight",
        "description": "Dry weight estimation accounting for edema or ascites."
    },
    {
        "name": "IBW with Amputation",
        "code": "ibw_amputation",
        "function_path": "nutrition.utils.ibw_amputation",
        "description": "IBW adjusted for amputation percentage."
    },
    {
        "name": "Waist to Height Ratio",
        "code": "waist_height_ratio",
        "function_path": "nutrition.utils.waist_height_ratio",
        "description": "Waist circumference / height, with gender-based risk interpretation."
    },
    {
        "name": "Percent Usual Body Weight",
        "code": "percent_ubw",
        "function_path": "nutrition.utils.percent_ubw",
        "description": "Current weight as % of usual body weight."
    },
    {
        "name": "Percent Weight Change",
        "code": "percent_weight_change",
        "function_path": "nutrition.utils.percent_weight_change",
        "description": "Percentage weight loss from usual weight."
    },
    {
        "name": "Ideal Body Weight (IBW) For Spinal Cord Injury",
        "code": "ibw_sci",
        "function_path": "nutrition.utils.ibw_spinal_cord_injury",
        "description": "Mueller, C. M. (Ed.). (2024). The essential pocket guide for clinical nutrition (4th ed., p. 9). Academy of Nutrition and Dietetics."
    },
    {
        "name": "Demi-span Height Estimate",
        "code": "demi_span_height",
        "function_path": "nutrition.utils.demi_span_height",
        "description": "Height estimation from demi-span and gender."
    },
    {
        "name": "Knee Height Estimate",
        "code": "knee_height_estimate",
        "function_path": "nutrition.utils.knee_height_estimate",
        "description": "Height estimation using knee height and age."
    },
    {
        "name": "Harris-Benedict Equation",
        "code": "harris_benedict",
        "function_path": "nutrition.utils.harris_benedict",
        "description": "Basal metabolic rate based on gender, weight, height, and age."
    },
    {
        "name": "Mifflin-St Jeor Equation",
        "code": "mifflin_st_jeor",
        "function_path": "nutrition.utils.mifflin_st_jeor",
        "description": "BMR calculation widely used in clinical practice."
    },
    {
        "name": "Total Energy Expenditure",
        "code": "total_energy_expenditure",
        "function_path": "nutrition.utils.total_energy_expenditure",
        "description": "TDEE = BMR × physical factor × stress factor."
    },
    {
        "name": "Penn State Equation",
        "code": "penn_state",
        "function_path": "nutrition.utils.penn_state",
        "description": "Energy expenditure equation for ventilated patients (two variants)."
    },
    {
        "name": "Cunningham Equation",
        "code": "cunningham",
        "function_path": "nutrition.utils.cunningham",
        "description": "Energy expenditure using lean body mass."
    },
    {
        "name": "Baseline Fluid (BSA method)",
        "code": "baseline_fluid_bsa",
        "function_path": "nutrition.utils.baseline_fluid_bsa",
        "description": "Fluid needs estimation using body surface area rules."
    },
    {
        "name": "Baseline Fluid (Standard method)",
        "code": "baseline_fluid_standard",
        "function_path": "nutrition.utils.baseline_fluid_standard",
        "description": "Fluid needs estimation using standard pediatric/adult rules."
    },
    {
        "name": "Insulin-to-Carb Ratio",
        "code": "insulin_to_carb_ratio",
        "function_path": "nutrition.utils.insulin_to_carb_ratio",
        "description": "Rule of 500 (rapid) or 450 (short) to calculate carb ratio."
    },
    {
        "name": "Insulin Sensitivity Factor",
        "code": "insulin_sensitivity",
        "function_path": "nutrition.utils.insulin_sensitivity",
        "description": "Rule of 1800 (rapid) or 1500 (short) to calculate sensitivity."
    },
    {
        "name": "Initial Insulin Dose",
        "code": "insulin_initial_dose",
        "function_path": "nutrition.utils.insulin_initial_dose",
        "description": "Initial insulin dose estimate based on body weight."
    },
    {
        "name": "Free Water Deficit",
        "code": "free_water_deficit",
        "function_path": "nutrition.utils.free_water_deficit",
        "description": "Estimate water deficit based on sodium and TBW fraction."
    },
    {
        "name": "Nitrogen Balance",
        "code": "nitrogen_balance",
        "function_path": "nutrition.utils.nitrogen_balance",
        "description": "Protein metabolism status: anabolic or catabolic."
    },
    {
        "name": "Nutrition Risk Index",
        "code": "nutrition_risk_index",
        "function_path": "nutrition.utils.nutrition_risk_index",
        "description": "Nutritional risk assessment using albumin and IBW %."
    },
    {
        "name": "Total Lymphocyte Count",
        "code": "total_lymphocyte_count",
        "function_path": "nutrition.utils.total_lymphocyte_count",
        "description": "Immune function indicator based on WBC and lymphocyte %."
    },
    {
        "name": "Growth Velocity",
        "code": "growth_velocity",
        "function_path": "nutrition.utils.growth_velocity",
        "description": "Mueller, C. M. (Ed.). (2024). The essential pocket guide for clinical nutrition (4th ed., p. 62). Academy of Nutrition and Dietetics."
    },
    {
        "name": "Harris–Benedict (HBE)",
        "code": "HBE",
        "function_path": "nutrition.utils.harris_benedict",
        "description": ""
    },
    {
        "name": "Mifflin-St Jeor",
        "code": "BMR",
        "function_path": "nutrition.utils.mifflin_st_jeor",
        "description": ""
    },
    {
        "name": "Quick Method for Macronutrients & Fluid",
        "code": "Quick Method for Macronutrients & Fluid",
        "function_path": "nutrition.utils.quick_method",
        "description": ""
    },
    {
        "name": "Macronutrients Distribution",
        "code": "Macronutrients Distribution",
        "function_path": "nutrition.utils.macronutrient_distribution",
        "description": ""
    },
    {
        "name": "Recommended Dietary Allowance for Calories",
        "code": "RDA Calorie needs",
        "function_path": "nutrition.utils.rda_calories",
        "description": ""
    },
    {
        "name": "Recommended Dietary Allowance for Protein",
        "code": "RDA Protein needs",
        "function_path": "nutrition.utils.rda_protein",
        "description": ""
    },
    {
        "name": "Schofield ",
        "code": "schofield_bmr",
        "function_path": "nutrition.utils.schofield_bmr",
        "description": ""
    },
    {
        "name": "Catch-up Growth",
        "code": "catchup_growth",
        "function_path": "nutrition.utils.catchup_growth",
        "description": ""
    },
    {
        "name": "Gestation-Adjusted Age (Correction Age)",
        "code": "gestation_adjusted_age",
        "function_path": "nutrition.utils.gestation_adjusted_age",
        "description": ""
    },
    {
        "name": "Preterm Estimated Requirment ",
        "code": "preterm_estimated_requirement",
        "function_path": "nutrition.utils.preterm_estimated_requirement",
        "description": ""
    },
    {
        "name": " Ireton-Jones (Ventilator-Dependent)",
        "code": "ireton_jones_ventilator",
        "function_path": "nutrition.utils.ireton_jones_ventilator",
        "description": ""
    },
    {
        "name": "Ireton-Jones (Spontaneously Breathing)",
        "code": "ireton_jones_spontaneous",
        "function_path": "nutrition.utils.ireton_jones_spontaneous",
        "description": ""
    },
    {
        "name": "Curreri",
        "code": "curreri_burn",
        "function_path": "nutrition.utils.curreri_burn",
        "description": ""
    },
    {
        "name": "Energy Needs in Pregnancy by Age & BMI group",
        "code": "pregnancy_energy_needs",
        "function_path": "nutrition.utils.pregnancy_energy_needs",
        "description": ""
    },
    {
        "name": "Energy Needs in Pregnancy",
        "code": "pregnancy_simple_addition",
        "function_path": "nutrition.utils.pregnancy_simple_addition",
        "description": ""
    },
    {
        "name": "Energy needs in lactation",
        "code": "lactation_energy_needs",
        "function_path": "nutrition.utils.lactation_energy_needs",
        "description": ""
    },
    {
        "name": "Down Syndrome Calorie Requirement (Ages 5–12)",
        "code": "down_syndrome_calorie",
        "function_path": "nutrition.utils.down_syndrome_calorie",
        "description": ""
    },
    {
        "name": "Cerebral Palsy Calorie Requirement (Ages 5–12)",
        "code": "cerebral_palsy_calorie",
        "function_path": "nutrition.utils.cerebral_palsy_calorie",
        "description": ""
    },
    {
        "name": "Prader-willi Syndrome Calorie Requirement (Ages 5–12)",
        "code": "prader_willi_calorie",
        "function_path": "nutrition.utils.prader_willi_calorie",
        "description": ""
    },
    {
        "name": "Prognostic Nutrition Index",
        "code": "prognostic_nutrition_index",
        "function_path": "nutrition.utils.prognostic_nutrition_index",
        "description": ""
    },
    {
        "name": "Prognostic Inflammatory and Nutrition Index",
        "code": "prognostic_inflammatory_nutrition_index",
        "function_path": "nutrition.utils.prognostic_inflammatory_nutrition_index",
        "description": ""
    },
]

class Command(BaseCommand):
    help = "Seed all predefined equations into the database"

    def handle(self, *args, **kwargs):
        for eq in EQUATIONS:
            Equation.objects.update_or_create(
                code=eq["code"],
                defaults=eq
            )
        self.stdout.write(self.style.SUCCESS("✅ Equations seeded successfully"))
